### DISCUSSION TITLE
ImageSharp Crop function

### DISCUSSION THREAD

**Discussion by qlyu001** [SixLabors/ImageSharp#1325] [Q&A] [answered]
I was reading the documentation for Image Sharp. I only see the input parameter as rectangle. Can Image Sharp crop a polygon from an image as well?

**Comment by tocsoft** [comment]
Not directly but you can use a combination of the simple rectangular Crop, [`ImageBrush`](https://docs.sixlabors.com/api/ImageSharp.Drawing/SixLabors.ImageSharp.Drawing.Processing.ImageBrush.html?q=imagebrush) and a polygon to simulate the effect.

Workflow would be;
1. Create a shape for your area of interest
2. Due to the fact image brush renders as texture  you then crop your source image to the bounding box of the shape
3. create a new blank image the size of your shapes bounding box
4. translate the shape so the origin of the bounding box is 0,0
5. `Fill` the shape with the cropped source image onto the new blank image
6. Save the new image 
7. Profit ???


Let me know if you struggle and i'll try and get some code sample demoing the workflow.

**Comment by qlyu001** [comment, marked as answer]
Sounds good!  Thank you for the swift response!
